### PR TITLE
feat: Release framework with no link to UIKit

### LINKED ISSUE #3768: Ship an XCFramework without UIKit linked
### Problem Statement

At Dropbox, we currently build Sentry both with and without UIKit.  We require that Sentry in our File Provider extension not import UIKit because the UIKit symbols take up too much space in memory with FP's super low memory limitations.

We'd love to simplify our build by using your new XCFramework distributions instead, but we need both the ones you're releasing now and also ones built without the UIKit link in order to switch over.

### Solution Brainstorm

Build Sentry without a link to UIKit and provide those XCFramework(s) as well in your releases page.

### Are you willing to submit a PR?

Maybe

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Creates a framework with no link to UIKit to distribute as a GH release asset.

## :bulb: Motivation and Context

closes #3768

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@
 
 - Don't run onCrashedLastSession for nil Events (#3785)
 
+### Features
+
+- Release framework with no link to UIKit (#3793)
```
Comment (philipphofmann) [inline CHANGELOG.md:11]: `l`:
```suggestion
- Release framework without to UIKit/AppKit (#3793)
```

File: scripts/build-xcframework.sh [hunk 2]
```diff
@@ -66,3 +67,5 @@ generate_xcframework "Sentry" "-Dynamic"
 generate_xcframework "Sentry" "" staticlib
 
 generate_xcframework "SentrySwiftUI"
+
+generate_xcframework "Sentry" "-NoUIKit" mh_dylib Release_without_UIKit
```
Comment (philipphofmann) [inline scripts/build-xcframework.sh:71]: `m`: I think that's a bit cleaner
```suggestion
generate_xcframework "Sentry" "-WihoutUIKitOrAppKit" mh_dylib Release_without_UIKitOrAppKit
```

**Review by philipphofmann — COMMENTED** [review]
We don't have UIKit on macOS, but instead, we have AppKit. Our SDK includes headers to Cocoa.h, but we only use NSApplication notification names. Therefore, we should be able to replace all Cocoa.h imports with `#import <AppKit/AppKit.h>`. So, let's reflect that in the framework name, as pointed out in the comments.

File: scripts/build-xcframework.sh (hunk 2 above)
Comment (brustolin) [reply scripts/build-xcframework.sh:71]: The forth parameter is the configuration name. It cannot be changed

Comment (philipphofmann) [reply scripts/build-xcframework.sh]: I thought that's how the XCFramework is called. I only care about the name of the XCFramework. How is that one called then?

Comment (brustolin) [reply scripts/build-xcframework.sh]: Second parameter.
It will be `Sentry-WihoutUIKitOrAppKit`

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,6 +13,10 @@ more about how to use the Metrics API.
 - Don't run onCrashedLastSession for nil Events (#3785)
 - Redistributable static libraries should never be built with module debugging enabled (#3800)
 
+### Features
```
Comment (philipphofmann) [inline CHANGELOG.md:16]: There is an already existing features header in the current unreleased section. I believe this needs to move up a couple of lines.

**Review by philipphofmann — APPROVED** [review]
LGTM

**Pull request merged**